{"pageProps":{"questions":[{"id":"NDZESpRtMCs8VmGZ3mEL","answer_ET":"A","answer":"A","answer_description":"","url":"https://www.examtopics.com/discussions/cisco/view/60282-exam-350-901-topic-1-question-23-discussion/","answers_community":["A (100%)"],"timestamp":"2021-08-22 21:25:00","question_images":[],"unix_timestamp":1629660300,"discussion":[{"content":"Selected Answer: A\nThe correct answer is A. git reset --hard f414f31.\n\nThis command will reset the current branch to the specified commit (in this case f414f31), discarding all changes made since that commit. The --hard option tells Git to reset both the index and the working directory to match the specified commit.","comment_id":"876790","upvote_count":"2","timestamp":"1729540200.0","poster":"tartarus23"},{"upvote_count":"1","comment_id":"581891","content":"A is correct answer","poster":"rajesh034","timestamp":"1696604760.0"},{"poster":"duracell","comment_id":"514143","content":"Selected Answer: A\nA is correct.","timestamp":"1688132220.0","upvote_count":"3"},{"content":"Correct answer","poster":"lente_kriebels","timestamp":"1677101100.0","upvote_count":"1","comment_id":"429476"}],"choices":{"B":"git reset checkout --hard f414f31","A":"git reset --hard f414f31","D":"git checkout f414f31","C":"git reset --soft f414f31"},"isMC":true,"exam_id":329,"answer_images":[],"topic":"1","question_text":"Which Git command enables the developer to revert back to f414f31 commit to discard changes in the current working tree?","question_id":146},{"id":"qPL9lxkP7q3VZFBxXcAS","topic":"1","question_text":"Where must the data be encrypted to ensure end-to-end encryption when using an API?","discussion":[{"content":"Selected Answer: D\nit must be D.","poster":"lznlxl","timestamp":"1721178660.0","comment_id":"778502","upvote_count":"6"},{"timestamp":"1720268100.0","comment_id":"767784","upvote_count":"5","poster":"fb48","content":"D. on the source device before transmission"}],"timestamp":"2023-01-06 15:15:00","isMC":true,"url":"https://www.examtopics.com/discussions/cisco/view/94178-exam-350-901-topic-1-question-230-discussion/","unix_timestamp":1673014500,"answer_images":[],"question_images":[],"choices":{"D":"on the source device before transmission","B":"on the end device after a request is received","C":"on the device that consumes the API","A":"on the server that stores the data"},"answers_community":["D (100%)"],"answer_ET":"D","answer":"D","answer_description":"","question_id":147,"exam_id":329},{"id":"YTNNmx9yvOInyhRXScLu","question_text":"//IMG//\nRefer to the exhibits. An engineer writes a script to retrieve data from a REST API and must build support for cases where the response that contains data from the server may take a longer time than normal. Which code must be added to the snippet where the code is missing to catch such a timeout?\n\n//IMG//","timestamp":"2023-01-02 10:52:00","choices":{"A":"requests.exceptions.HTTPError","C":"requests.exceptions.ConnectTimeout","B":"requests.exceptions.DataTimeout","D":"requests.exceptions.ReadTimeout"},"answer_ET":"D","isMC":true,"question_id":148,"discussion":[{"timestamp":"1708016640.0","comment_id":"809775","upvote_count":"5","poster":"alexgrt","content":"Selected Answer: C\nI'd pick ConnectTimeout because the request is being retried"},{"upvote_count":"2","poster":"mimimi","timestamp":"1726142640.0","comment_id":"1005711","content":"Honestly the only one that makes more sense would be A. By default, requests do not have a timeout unless you specify one explicitly. Therefore, since you didn't defined the timeout in first place, you would never have a Connect or Read timeout. This is messed up.\n\"Most requests to external servers should have a timeout attached, in case the server is not responding in a timely manner. By default, requests do not time out unless a timeout value is set explicitly. Without a timeout, your code may hang for minutes or more.\"\nhttps://requests.readthedocs.io/en/latest/user/advanced/#timeouts"},{"timestamp":"1717043220.0","upvote_count":"4","poster":"whipmuffin","content":"Selected Answer: D\nNot sure?\nAsked an AI:\n```\n**requests.exceptions.ConnectTimeout** exception is raised when the client fails to establish a connection to the server within the specified time³. \n\nOn the other hand, **requests.exceptions.ReadTimeout** exception is raised when the server fails to send data in the specified time¹. \n\nIn your case, since you are waiting for a response from the server, **requests.exceptions.ReadTimeout** exception should be used to handle long response time from REST API¹. \n```","comment_id":"909879"},{"content":"Selected Answer: D\nfrom the first snapshot, readtimeout says \"the server did not send any data in the alloted amount of time\". So d is correct answer.","upvote_count":"1","poster":"jithin1234","comment_id":"884345","timestamp":"1714396500.0"},{"upvote_count":"4","content":"Selected Answer: D\nThe ReadTimeout exception is raised when the server has not sent a response during the timeout period, which is useful for handling cases where the response takes longer than expected to arrive. Option A, requests.exceptions.HTTPError, is raised when an invalid HTTP response is received, such as a 404 Not Found error. Option B, requests.exceptions.DataTimeout, is not a valid exception in the requests library. Option C, requests.exceptions.ConnectTimeout, is raised when the connection to the server cannot be established within the timeout period.","poster":"xChdjdufurpew","timestamp":"1708700100.0","comment_id":"819319"},{"poster":"netzbus","content":"Answer D","timestamp":"1704189120.0","comment_id":"763626","upvote_count":"2"}],"exam_id":329,"question_images":["https://img.examtopics.com/350-901/image2.png","https://img.examtopics.com/350-901/image3.png"],"answer":"D","answers_community":["D (64%)","C (36%)"],"unix_timestamp":1672653120,"answer_description":"","url":"https://www.examtopics.com/discussions/cisco/view/93548-exam-350-901-topic-1-question-231-discussion/","answer_images":[],"topic":"1"},{"id":"4cuh1qt5LpJWgzySzZsZ","unix_timestamp":1673014920,"answers_community":[],"discussion":[{"upvote_count":"4","timestamp":"1724417940.0","content":"In Python, sys.exit() is a function from the sys module that is used to exit the current process. When sys.exit() is called with an argument of 0, it means that the program terminated successfully without any errors, and the exit status returned to the parent process is 0. On the other hand, when sys.exit() is called with an argument of 1 or any non-zero value, it means that the program terminated with an error, and the exit status returned to the parent process is non-zero.\n\nIn general, it's a convention to return an exit status of 0 for a successful termination and a non-zero exit status for an error termination. This convention allows the calling process to know if the called process completed successfully or not. The specific non-zero exit status can also be used to indicate different types of errors, with different exit statuses assigned to different types of errors.","poster":"xChdjdufurpew","comment_id":"819327"},{"poster":"fb48","comment_id":"767789","content":"Correct answer","timestamp":"1720268520.0","upvote_count":"4"}],"question_id":149,"timestamp":"2023-01-06 15:22:00","question_text":"DRAG DROP\n-\n\nA developer is creating a Python script to analyze errors during REST API call operations. The script will be used with Cisco solutions and devices. Drag and drop the code from the bottom to the box where the code is missing to implement control flow for handling unrecoverable REST API calls. Not all options are used.\n\n//IMG//","exam_id":329,"answer":"","isMC":false,"url":"https://www.examtopics.com/discussions/cisco/view/94179-exam-350-901-topic-1-question-232-discussion/","answer_images":["https://img.examtopics.com/350-901/image4-1.png"],"question_images":["https://img.examtopics.com/350-901/image4.png"],"answer_description":"","answer_ET":"","topic":"1"},{"id":"LEDqzHL2Tath2cJXQXqQ","topic":"1","question_text":"A developer has issued the git add file1 file2 test.py command to add three files for the next commit, but then decides to exclude test.py from this commit. Which command needs to be used to exclude test.py from this commit but keep the rest of the files?","timestamp":"2023-01-02 10:51:00","discussion":[{"comment_id":"1111345","upvote_count":"2","timestamp":"1719844260.0","poster":"Ch33s3y","content":"Selected Answer: D\nD is correct"},{"content":"Answer is correct","timestamp":"1704194040.0","upvote_count":"1","poster":"ThePokemonNuzlockingMaster","comment_id":"940697"},{"poster":"netzbus","upvote_count":"4","timestamp":"1688284260.0","comment_id":"763624","content":"Answer D"}],"url":"https://www.examtopics.com/discussions/cisco/view/93547-exam-350-901-topic-1-question-233-discussion/","isMC":true,"unix_timestamp":1672653060,"answer_images":[],"question_images":[],"choices":{"D":"git reset -- test.py","B":"git clean -- test.py","A":"git stash -- file1 file 2","C":"git checkout -- file1 file2"},"answers_community":["D (100%)"],"answer_ET":"D","answer":"D","question_id":150,"answer_description":"","exam_id":329}],"exam":{"provider":"Cisco","isImplemented":true,"id":329,"isBeta":false,"lastUpdated":"12 Apr 2025","name":"350-901","numberOfQuestions":383,"isMCOnly":false},"currentPage":30},"__N_SSP":true}